### ISSUE TITLE
concurrent-ruby gem pin is not precise enough

### ISSUE THREAD

**Issue by jordansissel** [elastic/logstash#4141]
```
# From a plugin
% bundle exec rspec
RuntimeError: Logstash expects concurrent-ruby version 0.9.1 and version 0.9.2 is installed, please verify this patch: /Users/
jls/.rvm/gems/jruby-1.7.22/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb
```

The problem is that `bundle install` fetched 0.9.2 because our dependency on `~> 0.9.1` and our concurrent ruby warning thingy patch barfs if we are using anything but 0.9.1

**Comment by jordansissel** [comment]
(this only affects master, really)

**Comment by jordansissel** [comment]
I was going to send a patch, but I'm not entirely sure at a glance if we should pin exactly 0.9.1 (in gem dependency) or relax the silence_concurrent_ruby patch to support 0.9.x

**Comment by jordansissel** [comment]
Over to @ph for some insight (based on authorship of b863ace5)

**Comment by ph** [comment]
Oh oh :(

**Comment by ph** [comment]
When we merged the shutdown changes the dependencie for concurrent-ruby was also changed. I think there are really close to 1.0

**Comment by ph** [comment]
I will review if they changed their warnings.

**Comment by Russell-IO** [comment]
@jordansissel - this affects logstash 2.0.0 & 1.5.5 as far as I can tell.
root@ip-10-12-0-62:/var/log/logstash# /opt/logstash/bin/logstash agent -f /opt/logstash-configuration/green/conf.d/v2/output/apache/
RuntimeError: Logstash expects concurrent-ruby version 0.9.1 and version 0.9.2 is installed, please verify this patch: /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb:53
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0-java/lib/logstash/patches.rb:1
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0-java/lib/logstash/patches.rb:6
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0-java/lib/logstash/environment.rb:1
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/lib/bootstrap/environment.rb:48

RuntimeError: Logstash expects concurrent-ruby version 0.9.1 and version 0.9.2 is installed, please verify this patch: /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.5-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.5-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb:53
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.5-java/lib/logstash/patches.rb:1
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.5-java/lib/logstash/patches.rb:6
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-1.5.5-java/lib/logstash/environment.rb:1
  require at org/jruby/RubyKernel.java:1040
   (root) at /opt/logstash/lib/bootstrap/environment.rb:48

**Comment by Russell-IO** [comment]
issue seems to be in /opt/logstash/vendor/bundle/jruby/1.9/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb - raising an exception ..
Does logstash need this patch with concurrenty-ruby 0.9.2 ?

**Comment by purbon** [comment]
just hit this issue after rebasing #3583 from master.

**Comment by purbon** [comment]
@ph I think, the best for now would be fix the version. We can always update to a new one as soon as they release it.

**Comment by ph** [comment]
@jordansissel I suggest we use 0.9.2 and relax the fix to match 0.9.X.

This warning will go away in future `concurrent-ruby` version because they have changed their mind and they will keep supporting JDK7 see https://github.com/ruby-concurrency/concurrent-ruby/issues/435

So once 1.0 is released we can drop that patch.

**Comment by ph** [comment]
I also make the patch not applied and not raise anything if the version doesn't match.

**Comment by purbon** [comment]
+1 on fixing this as you proposed. I guess we can move forward and get this done soon. Otherwise all builds are broken for now.

**Comment by ph** [comment]
nevermind the patch isn't need https://github.com/ruby-concurrency/concurrent-ruby/pull/440 this is 0.9.2 release.

**Comment by purbon** [comment]
@ph can we make a PR out of it?

**Comment by ph** [comment]
@Russell-IO did you update the plugins in your installation before getting that error?
What packages are you using ?

**Comment by purbon** [comment]
@suyograo @ph we need to release a point release to get this fix upstream, hitting this issue when doing work in the twitter plugin. 

```
skywalker% bundle exec rspec
RuntimeError: Logstash expects concurrent-ruby version 0.9.1 and version 0.9.2 is installed, please verify this patch: /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb
           (root) at /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb:53
          require at org/jruby/RubyKernel.java:1072
           (root) at /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-2.0.0-java/lib/logstash/patches.rb:1
          require at org/jruby/RubyKernel.java:1072
           (root) at /Users/purbon/.rbenv/versions/jruby-1.7.2
```

**Comment by ph** [comment]
We will next week, I've release beats with a constraint on 0.9.1 this
should fix the case when people where updating their plugins.

On Fri, Nov 6, 2015 at 5:31 AM, Pere Urbón <notifications@github.com> wrote:

> @suyograo <https://github.com/suyograo> @ph <https://github.com/ph> we
> need to release a point release to get this fix upstream, hitting this
> issue when doing work in the twitter plugin.
>
> skywalker% bundle exec rspec
> RuntimeError: Logstash expects concurrent-ruby version 0.9.1 and version 0.9.2 is installed, please verify this patch: /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb
>            (root) at /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-2.0.0-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb:53
>           require at org/jruby/RubyKernel.java:1072
>            (root) at /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-2.0.0-java/lib/logstash/patches.rb:1
>           require at org/jruby/RubyKernel.java:1072
>            (root) at /Users/purbon/.rbenv/versions/jruby-1.7.2
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/issues/4141#issuecomment-154374363>.
>



-- 
--
ph
Software Engineer

**Comment by purbon** [comment]
@ph makes sense. I think a lesson to be learn form this, is be careful with open ranges in dependencies, even small ones can cause trouble :+1: thanks man!

**Comment by driskell** [comment]
I'm seeing this happen when plugins are updated.

If you run `bin/plugin update logstash-input-courier` for example, it ends up upgrading the `concurrent-ruby` gem too. Could it because a dependency on `logstash-core` means when updating a plugin it updates all involved dependencies too?

**Comment by driskell** [comment]
Any advice on how to fix an installation broken by this?

**Comment by driskell** [comment]
I managed to fix it by adding `gem "concurrent-ruby", "0.9.1"` to the `Gemfile` and then updating any plugin, such as `bin/plugin update logstash-input-tcp`. Even if the plugin doesn't update it correctly rolls back the `concurrent-ruby` gem

**Comment by ph** [comment]
@driskell thanks for following this up, you can can also do bin/plugin install --no-verify which should only update the added dependencies.
